### PR TITLE
Make assignment a statement in Syntax.md

### PR DESCRIPTION
**Pull request by jimman2003**
force to change line 709 to something kind of non sensical

### CODE REVIEW

File: docs/markdown/Syntax.md [hunk 1]
```diff
@@ -706,7 +706,7 @@ DECIMAL_NUMBER: /[1-9][0-9]*/
 dictionary_literal: "{" [key_value_list] "}"
 equality_expression: relational_expression | (equality_expression equality_operator relational_expression)
 equality_operator: "==" | "!="
-expression: assignment_expression
+expression: assignment_statement
 expression_list: expression ("," expression)*
```
Comment (dcbaker) [inline docs/markdown/Syntax.md:710]: This still doesn't seem right. Assignment is a statement, but an expression isn't assignment

**Review by dcbaker — CHANGES_REQUESTED** [review]

File: docs/markdown/Syntax.md (hunk 1 above)
Comment (jimman2003) [reply docs/markdown/Syntax.md:710]: I know ... could really think of a way to resolve this

File: docs/markdown/Syntax.md [hunk 2]
```diff
@@ -695,7 +695,7 @@ additive_expression: multiplicative_expression | (additive_expression additive_o
 additive_operator: "+" | "-"
 argument_list: positional_arguments ["," keyword_arguments] | keyword_arguments
 array_literal: "[" [expression_list] "]"
-assignment_expression: conditional_expression | (logical_or_expression assignment_operator assignment_expression)
+assignment_statement: conditional_expression | (logical_or_expression assignment_operator assignment_expression)
```
Comment (dcbaker) [inline docs/markdown/Syntax.md:698]: I think this should be `assignment_statement: IDENTIFIER assignment_operator expression`

File: docs/markdown/Syntax.md (hunk 1 above)
Comment (dcbaker) [reply docs/markdown/Syntax.md:710]: And this should be:
`expression: conditional_expression | (logical_or_expression assignment_operator assignment_expression)`

File: docs/markdown/Syntax.md (hunk 2 above)
Comment (dcbaker) [reply docs/markdown/Syntax.md:698]: Actually, meson++ has this https://github.com/dcbaker/meson-plus-plus/blob/b7099bd3e8b3d1a2a7c8fea63821993ccb7da7ff/src/frontend/parser.yy#L106,
so I think this should be
`assignment_statement: expression asssignment_operator expression`

**Review by dcbaker — APPROVED** [review]

**Pull request merged**